templates/_composition/CaliburnMicro/Feature.ShareTargetExamplePage/Views/wts.ItemNameExamplePage.xaml

```diff
@@ -0,0 +1,62 @@
+﻿<Page
+    x:Class="Param_ItemNamespace.Views.wts.ItemNameExamplePage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    Style="{StaticResource PageStyle}"
+    xmlns:cm="using:Caliburn.Micro"
+    xmlns:templateSelectors="using:Param_ItemNamespace.TemplateSelectors"
+    xmlns:models="using:Param_ItemNamespace.Models"
+    mc:Ignorable="d">
+    <Grid
+        x:Name="ContentArea">
+
+        <Grid.RowDefinitions>
+            <RowDefinition x:Name="TitleRow" Height="48"/>
+            <RowDefinition Height="*"/>
+            <RowDefinition Height="48"/>
+        </Grid.RowDefinitions>
+
+        <Grid Margin="{StaticResource MediumLeftRightMargin}">
+            <TextBlock
+                x:Name="TitlePage"
+                Text="{x:Bind ViewModel.PageTitle}"
+                Style="{StaticResource PageTitleStyle}" />
+        </Grid>
+
+        <ContentControl Grid.Row="1" Content="{x:Bind ViewModel.SharedData, Mode=OneWay}">
+            <ContentControl.ContentTemplateSelector>
+                <templateSelectors:SharedContentTemplateSelector>
```

Comment:
I tend to stay away from template selectors in Caliburn.Micro as the framework has better support built in. Take a look how the tabbed pivot locates different views for the view models. 

If you don't want to create different view models per share type (which IMO makes sense rather than a hold all `SharedContentModel`) then you can use the `DataFormat` as a context for view location much like master details does.

Reply:
I think I don't understand this. The idea is to have the same ViewModel and show different layout (in the same Page) depending on the kind of data that is shared with the application.

Reply:
This might be a subjective design decision, but a single view model for all share types feels a bit off. normally I'd recommend a `SharedViewModelBase` that holds common properties such as `Title` and other metadata with a `SharedWebLinkViewModel` inheriting from that.

This way the view only needs to contain the following xaml. CM will then find the appropriate view (in the above case `SharedWebLinkView`) and insert it  This nicely separates the display of the different types share data from each other. It also means we won't need the template selector.

``` xml
<ContentControl cm:View.Model="{Binding SharedData}" />
```

If you want to stick with a single view model for all share data then I'd recommend something like the following in the xaml.

``` xml
<ContentControl cm:View.Model="{Binding SharedData}" cm:View.Context="{Binding SharedData.DataFormat}" />
```

This affects the view location with a context on the format so if the data format was WebLink then the inserted view would be something like `/Views/SharedData/WebLink.xaml`. Again separating the views and discarding the template selector.